public/app/plugins/datasource/loki/APP_PLUGIN_DOCUMENTATION.md

```diff
@@ -0,0 +1,148 @@
+# Loki data source documentation for app plugins developers
+
+Welcome to the developer documentation for the Loki data source! This documentation is designed to assist app plugin developers in leveraging the power of Loki data source within Grafana. Loki data source provides a set of APIs to simplify common tasks, making it easier to create engaging app plugins.
+
+## Introduction
+
+Loki data source provides a variety of methods, but not all of them are suitable for external use. In this documentation, we will focus on the key APIs that are highly recommended for app plugin development.
+
+It's important to note that the other APIs and methods that are not mentioned here may undergo changes or may not be suitable for external integration. Therefore, we do not recommend relying on them for your development needs.
+
+## Recommended APIs
+
+e strongly advise using these recommended APIs instead of direct proxy calls because they offer several benefits:
+
+- Caching of results: These APIs intelligently cache results based on their input arguments and the currently selected time range. This caching mechanism improves performance by reducing redundant fetch requests.
+
+- Proper authentication handling: The recommended APIs correctly handle custom authentication data source options, ensuring seamless authentication when making requests. This enhances both security and data access control.
+
+- Improved instrumentation: By opting for these recommended APIs, you gain access to advanced instrumentation features. These features provide better monitoring and analysis capabilities, offering deeper insights into your data source interactions.
```

Comment:
How would an external developer use this instrumentation? Honest question, because I can't imagine.

Reply:
If app developer has an access to server logs, then they can see in more details why are their resource calls/requests failing, as those calls have better instrumentation (metrics/logs/traces). So for example, if some user says that they don't see available labels in app plugin, plugin developer can go do server logs and find logs and traces for when user was trying to fetch label names. And we have a significantly improved instrumentation for requests done trough `callResource` and `queryData`.

Reply:
Can we add at least a brief mention on how to access it? If out out scope, no problem.

Reply:
I added a note https://github.com/grafana/grafana/pull/76067/commits/c82321a34a25cac358f2c33fe8e8ce82d2bb37ad